### PR TITLE
add gtags support for c-c++ layer

### PR DESCRIPTION
**Pull request by chengyi**
add gtags support for c-c++ layer using `counsel-gtags`

### CODE REVIEW

File: layers/+tags/gtags/packages.el [hunk 1]
```diff
@@ -14,6 +14,7 @@
   '(
     ggtags
     (helm-gtags :requires helm)
+    (counsel-gtags :toggle (configuration-layer/layer-usedp 'ivy))
```
Comment (TheBB) [inline layers/+tags/gtags/packages.el]: Please use the new `:requires` keyword for this sort of thing. (See line above.)

File: layers/+lang/c-c++/packages.el [hunk 2]
```diff
@@ -174,3 +175,7 @@
     :post-init
     (dolist (mode '(c-mode c++-mode))
       (spacemacs/setup-helm-cscope mode))))
+
+(defun c-c++/post-init-counsel-gtags ()
+  (spacemacs/counsel-gtags-define-keys-for-mode 'c-mode)
+  (spacemacs/counsel-gtags-define-keys-for-mode 'c++-mode))
```
Comment (TheBB) [inline layers/+lang/c-c++/packages.el]: Is there any reason why you have done this only for C/C++ and not for all the other modes where `spacemacs/helm-gtags-define-keys-for-mode` is used currently?

File: layers/+tags/gtags/funcs.el [hunk 3]
```diff
@@ -56,6 +56,19 @@ Otherwise does nothing."
...
+  (spacemacs/set-leader-keys-for-major-mode mode
+    "gC" 'counsel-gtags-create-tags
+    "gd" 'counsel-gtags-find-definition
+    "gr" 'counsel-gtags-find-reference
+    "gS" 'counsel-gtags-find-symbol
+    "gs" 'counsel-gtags-dwim
+    "gu" 'counsel-gtags-update-tags
+    "gf" 'counsel-gtags-find-file
+    "gn" 'counsel-gtags-go-forward
+    "gp" 'counsel-gtags-go-backward))
```
Comment (TheBB) [inline layers/+tags/gtags/funcs.el]: The differences between this and the helm version should be documented in the README.

Comment (chengyi) [inline layers/+tags/gtags/funcs.el]: OK, I'll modify the README. thanks.

File: layers/+lang/c-c++/packages.el (hunk 2 above)
Comment (chengyi) [inline layers/+lang/c-c++/packages.el]: @TheBB I'll add variable to `gtags` layer, so people can decide which mode that the want gtags enable.

**Comment by beta1440** [comment]
Maybe we could have a `spacemacs/gtags-define-keys-for-mode (gtags-backend mode)` function where the `gtags-backend` arguement specifies whether to define keys for `helm-gtags` or `counsel-gtags`.

**Pull request closed without merging**